File: Api/taskmanagmentApi/taskmanagmentApi/Models/TaskModel.cs
using System.ComponentModel.DataAnnotations;

namespace taskmanagmentApi.Models
{
    public class TaskModel
    {
        [Key]
        public int TaskId { get; set; }
        public int ProjectId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int StatusId { get; set; }
        public int PriorityId { get; set; }
        public int AssignedBy { get; set; }
        public int AssignedTo { get; set; }
        public DateTime Assigneddate { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Client/TaskManagment/src/app/interface/task.model.ts
// task-create.dto.ts
export interface TaskCreateDto {
  projectId: number;
  moduleId: number | null;
  title: string;
  description: string;
  assignedBy: number;
  assignedTo: number;
  statusId: number;
  priorityId: number;
  assignedDate: string;
  dueDate: string;
}


export interface TaskResponseDto extends TaskCreateDto {
  taskId: number;
  createdAt: string;
}
